python/pyiceberg/catalog/rest.py

```diff
@@ -210,8 +199,8 @@ def headers(self) -> Properties:
             "Content-type": "application/json",
             "X-Client-Version": __version__,
         }
-        if self.token:
-            headers[AUTHORIZATION_HEADER] = f"{BEARER_PREFIX} {self.token}"
+        if token := self.properties.get("token"):
```

Comment:
Should this be `self.property("token")`?

Reply:
I've added a docstring to the property function:
```python
def property(self, key: str) -> str:
    """Returns a property from the properties variable. If it doesn't exist, it will raise an error.

    Args:
        key: The key of the property

    Returns: The value of the property

    Raises:
        ValueError: When the property cannot be found, with a pointer on how to set the property.
    """
    if key not in self.properties:
        raise ValueError(
            f"{type(self).__name__} expects an {key} property. Please set in config or using environment variable PYICEBERG_CATALOG__{self.name.upper()}__{key.upper()}"
        )
    return self.properties[key]
```

A token isn't required, for example, when authentication is disabled on the rest catalog.

Reply:
Would it be better to allow passing a default? That way we don't need to bypass the `property` method for cases where it is optional.

Reply:
I'd rather not change the signature to `Optional[str]` because then we have to do yet another `None` check 🤔